prime-router/src/main/kotlin/Report.kt

```diff
@@ -968,13 +972,14 @@ class Report : Logging {
                     )
             }
             val retval = mutableListOf<ItemLineage>()
-            for (i in 0 until prevHeader.reportFile.itemCount) {
-                if (newLineages[i] == null)
+            // Note indeces start at 1
+            for (index in 1..prevHeader.reportFile.itemCount) {
+                if (newLineages[index] == null)
                     error(
                         "Unable to create parent->child lineage " +
-                            "${prevHeader.reportFile.reportId} -> $newChildReportId: missing lineage $i"
+                            "${prevHeader.reportFile.reportId} -> $newChildReportId: missing lineage $index"
                     )
-                retval.add(newLineages[i]!!)
+                retval.add(newLineages[index]!!)
```

Comment:
Should be able to remove the `!!` since there's a null check above

Reply:
No.  the contents can be nullable, so I get a syntax error if I remove it.

Reply:
I don’t like using the `!!` operator in production code. It would be better to check see if it’s null and return a human readable error message.

Reply:
I refactored this, but does not look any prettier.

Reply:
It's not about prettier. It's about the fact that the `!!` assertion is claiming "I know better than the compiler" and if somehow it ended up null anyway it throws a less intuitive and controllable error. Checking for null ahead of time and throwing a human readable error is more conducive to future maintenance.